src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_CAPACITY = 10;
+    private T[] array;
+    private int size;
 
+    public MateStack() {
+        array = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size == array.length) {
+            resize();
+        }
+        array[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        return returnValue();
     }
 
     public T pop() {
-        return null;
+        T lastElement = returnValue();
+        array[size - 1] = null;
+        size--;
+        return lastElement;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private T returnValue() {
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return array[size - 1];
+    }
+
+    private void resize() {
+        T[] bufferArray = array;
+        array = (T[]) new Object[array.length + (array.length >> 1)];
```

Comment:
what >> do?

Reply:
right shift divides the number by 2. So basically he is increasing array in 1.5 times.

Reply:
👍